src/shell/dotnet/Shell/MainWindow.xaml.cs

```diff
@@ -106,12 +106,24 @@ public ModuleViewModel(IModuleManifest manifest, ImageSourceProvider imageSource
             }
             else if (manifest.TryGetDetails<NativeManifestDetails>(out var nativeManifestDetails))
             {
-                var icon = System.Drawing.Icon.ExtractAssociatedIcon(Path.GetFullPath(nativeManifestDetails.Path.ToString()));
+                var icon = GetNativeAppIcon(nativeManifestDetails);
+                if (icon != null) { 
+                    ImageSource = System.Windows.Interop.Imaging.CreateBitmapSourceFromHIcon(
```

Comment:
Can we check whether we need to release unmanaged resources like handles after this?

Reply:
Good catch, an Icon is disposable

Reply:
It is unrelated to this change - I would say go ahead and merge this after switching to using and create a new item for to check for losing handles. Want to avoid having this open too long.